### PR TITLE
Lets lockers/crates protect food from ants

### PR DESCRIPTION
**Pull request by CinnamonSnowball**
## What Does This PR Do
-​ Makes closets protect food from ants
-​ Refactors the ants check
-​ Fixes space ants decals _protecting_ food from being infested, now they only prevent further ant decal spawns

## Why It's Good For The Game
Presumably, the fact that items that pop out from a locker when opened render on the floor is just an implementation detail, and realistically they're sitting on various shelves and whatnot, which ants should not be able to reach considering they can't scale tables or racks.

More practically, this gives chefs and any other locker food storage enthusiasts a little bit more leniency.

## Testing
-​ Spawned in 2 different lockers, 2 different tables, and a rack
-​ Opened one locker
-​ Put a bunch of cheeseburgers on top of the open locker, the table, the rack, on the floor covered by the closed locker, and also loosely on the floor alone
-​ Waited a while
-​ Made sure that the only the loose floor burgers got infested

## Changelog
:cl:
tweak: Lockers now protect food from space ants
fix: Physical space ants floor decals no longer protect food from being infested
/:cl:

### CODE REVIEW

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -388,6 +388,7 @@
 
 	return ..()
 
+
```
Comment (Henri215) [inline code/game/objects/structures/crates_lockers/closets.dm:391]:
```suggestion
```

Comment (CinnamonSnowball) [reply code/game/objects/structures/crates_lockers/closets.dm:391]: This is for readability, separating definitions of basetype closet procs and a subtype

File: code/modules/food_and_drinks/food_base.dm [hunk 2]
```diff
@@ -47,17 +62,27 @@
...
 		ant_location = T
+		return
+
+	var/protected = FALSE // Check if some object on our turf protects the food from ants
+	for(var/obj/structure/S in T)
+		if(is_type_in_typecache(S, ant_suppressors))
+			protected = TRUE
+			break
+	if(protected)
+		return
```
Comment (Contrabang) [inline code/modules/food_and_drinks/food_base.dm:81]: you can get rid of this protected var and just return
```suggestion
			return
```

File: code/modules/food_and_drinks/food_base.dm [hunk 3]
```diff
@@ -47,17 +62,27 @@
 /obj/item/reagent_containers/food/proc/check_for_ants()
 	last_ant_time = world.time
 	var/turf/T = get_turf(src)
```
Comment (Contrabang) [inline code/modules/food_and_drinks/food_base.dm:64]: Can you move this get_turf call down to below the `!isturf(loc)`?

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Review by Contrabang — APPROVED** [review]
LGTM

**Review by S34NW — APPROVED** [review]
ANTS!

**Pull request merged**